packages/client/src/request-login.test.ts

```diff
@@ -1,11 +1,108 @@
-import { getLoginUrlAndCodeVerifier } from './request-login';
+import { DEFAULT_SCOPE_STRING } from './constants';
+import { getLoginUrlWithCodeVerifierAndState } from './request-login';
 
-test('getLoginUrlAndCodeVerifier', async () => {
-  const { url } = await getLoginUrlAndCodeVerifier({
-    baseUrl: 'http://logto.dev/oidc/auth',
-    clientId: 'foo',
-    scope: 'openid offline_access',
-    redirectUri: 'http://localhost:3000',
+const BASE_URL = 'http://logto.dev/oidc/auth';
+const CLIENT_ID = 'foo';
+const REDIRECT_URI = 'http://localhost:3000';
+
+describe('getLoginUrlAndCodeVerifier', () => {
+  test('url must start with baseUrl', async () => {
+    const { url } = await getLoginUrlWithCodeVerifierAndState({
+      baseUrl: BASE_URL,
+      clientId: CLIENT_ID,
+      scope: DEFAULT_SCOPE_STRING,
+      redirectUri: REDIRECT_URI,
+    });
+    expect(url.startsWith(BASE_URL)).toBeTruthy();
+  });
+
+  test('url must contain code_challenge=EXPECTED_VALUE', async () => {
```

Comment:
can't see an expected code_challenge value in this test
check url contains code_challenge is enough

Reply:
Fixed above.

Here is a wrong test description.